Eigen fails to build on VS 2012 sp4 (cl version 17.00.61030)

See https://github.com/OpenMS/OpenMS/pull/651

Eigen 3.2 doesn't build on VS 2012 sp4 (cl version 17.00.61030). There is a fix up to sp3 in the current eigen development branch (see https://bitbucket.org/eigen/eigen/src/4206f0c33abe98e6d28d0fd660623d0279273cee/cmake/EigenDetermineVSServicePack.cmake?at=default)
I think we shouldn't proceed until this is resolved.

```
CMake Error at C:/Program Files (x86)/CMake 2.8/share/cmake-2.8/Modules/CMakeDetermineVSServicePack.cmake:128 (list):
  list index: 3 out of range (-3, 2)
Call Stack (most recent call first):
  cmake/EigenDetermineVSServicePack.cmake:9 (_DetermineVSServicePack_CheckVersionWithTryCompile)
  cmake/EigenTesting.cmake:316 (EigenDetermineVSServicePack)
  cmake/EigenTesting.cmake:428 (ei_get_compilerver)
  cmake/EigenConfigureTesting.cmake:8 (ei_set_build_string)
  CMakeLists.txt:349 (include)
```

--- a/libraries.cmake/eigen.cmake
+++ b/libraries.cmake/eigen.cmake
@@ -14,8 +14,14 @@ macro( OPENMS_CONTRIB_BUILD_EIGEN )
   endif()
   OPENMS_SMARTEXTRACT(ZIP_ARGS ARCHIVE_EIGEN "EIGEN" "CMakeLists.txt")
 
-  # eigen doesn't allow insource builds
+	if(MSVC)
+    # fixes MSVC 2012 compiler detection
+    set(_PATCH_FILE "${PATCH_DIR}/eigen/EigenDetermineVSServicePack.cmake.patch")
+    set(_PATCHED_FILE "${EIGEN_DIR}/cmake/EigenDetermineVSServicePack.cmake")
+    OPENMS_PATCH( _PATCH_FILE EIGEN_DIR _PATCHED_FILE)
+  endif()
 
+  # eigen doesn't allow insource builds
   set(_EIGEN_BUILD_DIR "${EIGEN_DIR}/build")
   file(TO_NATIVE_PATH "${_EIGEN_BUILD_DIR}" _EIGEN_NATIVE_BUILD_DIR)
 

--- a/patches/eigen/EigenDetermineVSServicePack.cmake.patch
+++ b/patches/eigen/EigenDetermineVSServicePack.cmake.patch
@@ -1,0 +1,36 @@
+--- cmake/EigenDetermineVSServicePack.cmake
++++ cmake/EigenDetermineVSServicePack.cmake
+@@ -4,19 +4,24 @@
+ # _DetermineVSServicePack_FastCheckVersionWithCompiler which lead to errors on some systems.
+ function(EigenDetermineVSServicePack _pack)
+     if(NOT DETERMINED_VS_SERVICE_PACK OR NOT ${_pack})
+-
+-        if(NOT DETERMINED_VS_SERVICE_PACK)
+-            _DetermineVSServicePack_CheckVersionWithTryCompile(DETERMINED_VS_SERVICE_PACK _cl_version)
+-            if(NOT DETERMINED_VS_SERVICE_PACK)
+-                _DetermineVSServicePack_CheckVersionWithTryRun(DETERMINED_VS_SERVICE_PACK _cl_version)
+-            endif()
+-        endif()
+-
+         if(DETERMINED_VS_SERVICE_PACK)
+-
+             if(_cl_version)
+                 # Call helper function to determine VS version
+                 _DetermineVSServicePackFromCompiler(_sp "${_cl_version}")
++
++                # temporary fix, until CMake catches up
++                if (NOT _sp)
++                    if(${_cl_version} VERSION_EQUAL "17.00.50727.1")
++                        set(_sp "vc110")
++                    elseif(${_cl_version} VERSION_EQUAL "17.00.51106.1")
++                        set(_sp "vc110sp1")
++                    elseif(${_cl_version} VERSION_EQUAL "17.00.60315.1")
++                        set(_sp "vc110sp2")
++                    elseif(${_cl_version} VERSION_EQUAL "17.00.60610.1")
++                        set(_sp "vc110sp3")
++                    endif()
++                endif()
++
+                 if(_sp)
+                     set(${_pack} ${_sp} CACHE INTERNAL
+                         "The Visual Studio Release with Service Pack")

